### ISSUE TITLE
Comparison option "Show key-values with different content only" doesn't work

### ISSUE THREAD

**Issue by Philippe85** [Azure/AppConfiguration#457]
In Azure Portal, in App Configuration ressource comparison of keys, the option "Show key-values with different content only" doesn't work. All data are displayed.
So, the comparison of key-values between to label is impossible !

**Comment by jiayi11** [comment]
Just want to clarify, are you trying to compare key-values with same values but different labels? If so, then this is expected because the option "Show key-values with different content only" will only filter out those with exact same key, value, label and content type.

**Comment by Philippe85** [comment]
Yes, I want to compare values of the same keys in two labels differents.
It is the goal, to compare values between two environment (two labels) and check if values are correct.
If we can't that, what is the interest of this comparison ?

**Comment by jiayi11** [comment]
The compare blade is designed to show users all differences between key-values include labels. We also add an info icon in front of all differences and hopefully it can help you catch the different values easier. 
![image](https://user-images.githubusercontent.com/14067510/108259753-5a1cfb00-7116-11eb-87b5-9914c7df73f7.png)

**Comment by Philippe85** [comment]
The option "Show key-values with different content only", as its name indicates, is only for keys and values, not for labels.

If we compare the keys and values between 2 different labels (known in advance) with the option "Show key-values with different content only", we expect to see only lines with different keys or different values. If you also compare the labels, then the option doesn't work any more because you end up with all the lines (with all key-values without filter).

If we want all the key-values, then we just have to uncheck the option.

**Comment by zhenlan** [comment]
@Philippe85 the scenario you described makes sense to me. If I understand you correctly, you expect, in my example below, *mykey* does not show up in the comparison list because it has the same value for the two labels (*dev* and *test*) I am comparing.

![image](https://user-images.githubusercontent.com/10566826/111849027-e4997b80-88c9-11eb-93d3-9bca682edcc1.png)

**Comment by Philippe85** [comment]
Yes. It is exact. Not show keys with same values. Show only keys with differences.

**Comment by Philippe85** [comment]
The options "Hide matching key-value pairs" and "Hide all matching keys" are incomprehensible.
What is the meaning of "matching" here !

**Comment by cablooshe** [comment]
"Matching" here is to describe the two compared elements having the same fields (so both the same key and value field for "Hide matching key-value pairs", or only the same key in "Hide all matching keys"). "Matching" does not consider differences in labels.

So to compare to the example above that Zhenlan has, "Hide matching key-value pairs" will remove _mykey_from view, while "Hide all matching keys" will remove both _mykey_ and _mykey2_.

**Comment by jiayi11** [comment]
We already have the option "Hide matching key-value pairs" to achieve it. I think the issue left now is the compare options are not easy to understand. 

> The options "Hide matching key-value pairs" and "Hide all matching keys" are incomprehensible.

I'll follow up with the options naming.
